hack/test-olm.sh

```diff
@@ -0,0 +1,285 @@
+#!/bin/bash
+
+set -ex
+
+OPT=${1:-kind}
+
+if [[ $OPT != "kind" ]] && [[ $OPT != "minikube" ]]; then
+	echo "error: argument must be either kind or minikube"
+	exit 1
+fi
+
+LOCAL_OLM=olm-repo
+LOCAL_OPREG=operator-registry
+
+case "$OPT" in
+	minikube)
+		LOCAL_REGISTRY=localhost.localdomain:5000
+		HOST_IP=$(hostname -I | awk '{ print $1 }')
+		;;
+
+	kind)
+		LOCAL_REGISTRY=localhost:5000
+		;;
+esac
+
+NMOBUNDLE=node-maintenance-operator-bundle
+NMOREG=node-maintenance-operator-registry
+NMO=node-maintenance-operator
+CATALOG_NAMESPACE="olm"
+REGISTRY=quay.io/kubevirt
+
+log_to_file() {
+	exec &> >(tee -a test-olm.out)
+}
+
+start_registry() {
+	hreg_name='local-registry'
+	reg_port='5000'
+	running="$(docker inspect -f '{{.State.Running}}' "${reg_name}" 2>/dev/null || true)"
+	if [ "${running}" != 'true' ]; then
+	  docker run \
+		-d --restart=always -p "${reg_port}:5000" --name "${reg_name}" \
+		registry:2
+	fi
+}
+
+setup_src() {
+	if [[ ! -f $LOCAL_OLM/repo_init ]]; then
+		git clone https://github.com/operator-framework/operator-lifecycle-manager.git $LOCAL_OLM
+		pushd $LOCAL_OLM
+		#git checkout origin/release-4.5 -b release-4.5
+		echo "" >repo_init
+		popd
+
+	fi
+
+	if [[ ! -f $LOCAL_OPREG/repo_init ]]; then
+		git clone https://github.com/operator-framework/operator-registry $LOCAL_OPREG
```

Comment:
What do you need this for?

Reply:
need to build and run the opm cli tool - that one is needed to build the database index from the bundle. https://docs.okd.io/latest/operators/understanding_olm/olm-understanding-olm.html#olm-upgrades_olm-understanding-olm

Reply:
same here: we don't need to build `opm` ourself. You can download it from https://github.com/operator-framework/operator-registry/releases

Reply:
done. now using a downloaded binary image for the utility.